src/components/NewMessage/NewMessagePollEditor.vue

```diff
@@ -143,16 +170,41 @@ export default {
 		async createPoll() {
 			const poll = await this.pollsStore.createPoll({
 				token: this.token,
-				question: this.pollQuestion,
-				options: this.pollOptions,
-				resultMode: this.isPrivate ? 1 : 0,
-				maxVotes: this.isMultipleAnswer ? 0 : 1
+				question: this.pollForm.question,
+				options: this.pollForm.options,
+				resultMode: this.pollForm.isPrivate ? 1 : 0,
+				maxVotes: this.pollForm.isMultipleAnswer ? 0 : 1
 			})
 			if (poll) {
 				this.dismissEditor()
 			}
 		},
 
+		triggerImport() {
+			this.$refs.pollImport.click()
+		},
+
+		importPoll(event) {
+			if (!event?.target?.files?.[0]) {
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e) => {
+				try {
+					const jsonObject = JSON.parse(e.target.result)
+					for (const key of Object.keys(this.pollForm)) {
+						if (jsonObject[key] !== undefined) {
+							this.pollForm[key] = jsonObject[key]
+						}
```

Comment:
I'd say if it is corrupted in any of its keys, we don't render it. We also need to show a hint indicating that the file is not Talk poll json file or has been modified.

Reply:
So we need some sort of a validation beforehand

Reply:
Yes, that are ONLY the keys we have from Poll are there (4)